src/storage/src/source/reclock.rs

```diff
@@ -573,19 +548,19 @@ where
             let binding_ts = if *self.upper == [IntoTime::minimum()] {
                 IntoTime::minimum()
             } else {
-                ts
+                binding_ts.clone()
```

Comment:
why does this need a `.clone()` here? Seems like we are consuming the original `binding_ts` value once

Reply:
It's because this is in a while loop and so we may need to use the binding_ts multiple times

Reply:
Ah my mistake, makes sense